### ISSUE TITLE
Side Sheet: Tab order should focus app content first

### ISSUE THREAD

**Issue by acdvorak** [material-components/material-components-web#3944]
## TL;DR

Putting Side Sheets first in the source order and repositioning them with CSS screws up tab order.
This is bad for accessibility.

We need to find a better way to style the scrim and app content elements that doesn't rely on the
CSS [`order` property](https://mdn.io/css/order) and [sibling combinators (`+` and `~`)](https://mdn.io/css/general_sibling_selectors).

This issue was first brought up in https://github.com/material-components/material-components-web/pull/3924#discussion_r226361478.

## Problem

The first iteration of Side Sheet (PR #3924) was forked from Drawer, which uses [sibling combinators (`+` and `~`)](https://mdn.io/css/general_sibling_selectors) to style the scrim and app content:

https://github.com/material-components/material-components-web/blob/4d73a7a3e9834caedc85579e36b9f26699185f52/packages/mdc-side-sheet/_mixins.scss#L70

https://github.com/material-components/material-components-web/blob/4d73a7a3e9834caedc85579e36b9f26699185f52/packages/mdc-side-sheet/_mixins.scss#L101

https://github.com/material-components/material-components-web/blob/4d73a7a3e9834caedc85579e36b9f26699185f52/packages/mdc-side-sheet/_mixins.scss#L230

That approach works fine for Drawer because Drawers are always on the _leading_ edge of the page (the _left_ side in LTR languages), which means it's safe to assume that the drawer element comes first in the [source order](http://adrianroselli.com/2015/10/html-source-order-vs-css-display-order.html). The visual placement of the Drawer matches the source/tab order, and life is good.

Side Sheets, on the other hand, are displayed on the _trailing_ edge of the page (the _right_ side in LTR languages). There's no such thing as a "reverse" sibling combinator, so like Drawer, the Side Sheet needs to come first in the source order for the styles to work.

This presents a problem for Permanent Side Sheets because the parent element of the Side Sheet and the app content is `display: flex`, so whichever element comes first in the source order will be displayed on the _leading_ edge of the page. To fix the _visual_ order, we can use the [CSS `order` property](https://mdn.io/css/order):

https://github.com/material-components/material-components-web/blob/4d73a7a3e9834caedc85579e36b9f26699185f52/packages/mdc-side-sheet/_mixins.scss#L126-L130

Unfortunately, `order` doesn't affect tab order, so pressing the <kbd>Tab</kbd> key will focus the side sheet before the app content, which is wrong.

**Comment by williamernest** [comment]
We could give guidance to users to include a skip link at the beginning of the `body`, or between the drawer and side sheet that allows users to skip directly to the `main` element. 

Example: https://codepen.io/williamernest/pen/jeXaGy

**Comment by bonniezhou** [comment]
Closing this as obsolete.
